### ISSUE TITLE
[Bug]: Error retrieving document: Error: Could not find Chrome (ver. 114.0.5735.90)

### ISSUE THREAD

**Issue by lifosmin** [puppeteer/puppeteer#10388]
### Bug expectation

I was trying to open a website for my scrapping that i hosted on cloud run, but the problem that i was encountering that when it run always gives me [Bug]: Error retrieving document: Error: Could not find Chrome (ver. 114.0.5735.90)

 

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Minimal, reproducible example

```TypeScript
import cheerio from "cheerio";
import puppeteer from 'puppeteer';
import randomUseragent from 'random-useragent';

export const scrapeData = async (productName) => {
  const product = productName.replace(/ /g, "+");

  console.log(1);
  const url = `https://www.tokopedia.com/search?st=product&q=${product}`;
  const randomAgent = randomUseragent.getRandom();
  const browser = await puppeteer.launch({
    headless: true,
    args: [
      '--no-sandbox',
      '--disable-setuid-sandbox',
    ],
  });
  console.log(2);

  const context = await browser.createIncognitoBrowserContext();
  const page = await context.newPage();
  await page.setJavaScriptEnabled(true);
  await page.setUserAgent("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.83 Safari/537.36");
  
  const domLoadedPromise = new Promise((resolve) => {
    page.once('load', resolve);
  });
  
    await page.goto(url, { waitUntil: 'networkidle0' });
    // await page.waitForSelector('[data-testid="master-product-card"]');
  await domLoadedPromise;


  const body = await page.evaluate(() => {
    return document.querySelector('body').innerHTML;
  });
  
  const $ = cheerio.load(body);
  const listItems = $('[data-testid="master-product-card"]');
  console.log(3);

  var result
  listItems.each(function (idx, el) {
    var nama = $('[data-testid="spnSRPProdName"]', el).text();
    var harga = $('[data-testid="spnSRPProdPrice"]', el).text();
    var image = $('[data-testid="imgSRPProdMain"]', el).attr("src");
    var link = $('a[href]', el).attr("href");
    if (harga != null && harga != "") {
        result = {
            "Treatment for": productName,
            "nama": nama,
            "harga": harga,
            "link": link,
            "image": image
          };

      return false
    }
    
  });

  console.log("Url: ", url);
  console.log("For This Product: ", productName);

  
  await browser.close();
  
  return result;
};
```

### Error string

Error retrieving document: Error: Could not find Chrome (ver. 114.0.5735.90)

### Puppeteer configuration

```TypeScript
# Use the official Node.js 14 image as the base
FROM node:18

# Set the working directory inside the container
WORKDIR /app

# Copy package.json and package-lock.json to the working directory
COPY package.json package-lock.json ./

# Install project dependencies
RUN npm install

# Install necessary dependencies for Puppeteer
RUN apt-get update \
    && apt-get install -y wget gnupg \
    && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
    && sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' \
    && apt-get update \
    && apt-get install -y google-chrome-stable fonts-ipafont-gothic fonts-wqy-zenhei fonts-thai-tlwg fonts-kacst fonts-freefont-ttf libxss1 \
      --no-install-recommends \
    && rm -rf /var/lib/apt/lists/*
# Copy the rest of the project files to the working directory
COPY . .

# Start the Express application
CMD [ "npm", "run", "dev" ]
```

### Puppeteer version

20.6.0

### Node version

18

### Package manager

npm

### Package manager version

18

### Operating system

Linux

**Comment by ThomasLarge** [comment]
I'm seeing this too, was working fine but just stopped working on a fresh install. Does this always pull the latest chrome? Seen issues with this and Cypress before

**Comment by lifosmin** [comment]
my package json looks like this, I'm sure the Puppeteer is installed

`{
    "name": "api",
    "version": "1.0.0",
    "description": "",
    "main": "server.js",
    "type": "module",
    "scripts": {
      "dev": "nodemon server.js"
    },
    "keywords": [],
    "author": "",
    "license": "ISC",
    "dependencies": {
      "body-parser": "^1.20.2",
      "cheerio": "^1.0.0-rc.12",
      "dotenv": "^16.1.4",
      "express": "^4.18.2",
      "firebase-admin": "^11.9.0",
      "mysql": "^2.18.1",
      "nodemon": "^2.0.22",
      "puppeteer": "^20.6.0",
      "puppeteer-core": "^20.7.1",
      "random-useragent": "^0.5.0"
    }
  }
  
`

**Comment by OrKoN** [comment]
We had no changes here recently. Puppeteer always pulls the pinned version of Chrome. Could you please share a reduced reproduction for this? Our official docker image builds and tests without issue https://github.com/puppeteer/puppeteer/pkgs/container/puppeteer

**Comment by OrKoN** [comment]
Update here: it might be that there is some problem with the server where the browser is downloaded from. Today we see occasional ECONNREFUSED errors, you should see it in the npm install log if the installation fails: https://github.com/puppeteer/puppeteer/actions/runs/5288559932/jobs/9570469267?pr=10396

**Comment by lifosmin** [comment]
the bug happened when i was trying to connect browser const browser = await puppeteer.launch({
    headless: true,
    args: [
      '--no-sandbox',
      '--disable-setuid-sandbox',
    ],
  });
  
  I configured docker file to install all dependencies that was needed to run at cloud.

**Comment by yoursunny** [comment]
I'm seeing Chromium download failure too.
https://github.com/yoursunny/NDNts/actions/runs/5291639615/jobs/9577451367#step:6:94

**Comment by lifosmin** [comment]
can i know how to resolve this?

**Comment by andyMrtnzP** [comment]
Seems like the default server the `browsers` package pulls Chrome from is not serving r`114.0.5735.133`. Has no issue serving, say [112.0.5615.49](https://chromedriver.storage.googleapis.com/112.0.5615.49/chromedriver_linux64.zip), but trying to download [114.0.5735.133](https://chromedriver.storage.googleapis.com/114.0.5735.133/chromedriver_linux64.zip) will fail.

**Comment by OrKoN** [comment]
It seems the server is healthy again and I could not see any failures in the last two days.

**Comment by abdalem** [comment]
@lifosmin HI ! Did you resolved your issue ?

**Comment by AdwaitMaidaan** [comment]
Having the same issue. I did all the steps from creating the config file to adding the launch parameters to puppeteer, but nothing seems to be working. 
I have the .cache directory with the chrome version inside it, but I still get the error "Error: Could not find Chrome (ver. 114.0.5735.133)"
Any help would be appreciated.

**Comment by joel-ghats** [comment]
@OrKoN  have  the same issue for me today, was able to launch chrome  successfully while container is running in local 

` const browser = await puppeteer.launch({
                executablePath: process.env.CHROME_BIN || null,
                args: ['--no-sandbox', '--headless', '--disable-gpu','--disable-setuid-sandbox'],
                headless: "new"
              });`

but when i try to launch it in google app engine , i am getting following error 



`INFO 2023-06-23T10:29:14.307788Z [pid1-nginx] Starting nginx (pid 28): /usr/sbin/nginx -c /tmp/nginxconf-1387706939/nginx.conf [session:D68SV90]
DEFAULT 2023-06-23T10:29:17.100363Z error Error: Could not find Chrome (ver. 114.0.5735.133). This can occur if either
DEFAULT 2023-06-23T10:29:17.100383Z 1. you did not perform an installation before running the script (e.g. `npm install`) or
DEFAULT 2023-06-23T10:29:17.100391Z 2. your cache path is incorrectly configured (which is: /root/.cache/puppeteer).
DEFAULT 2023-06-23T10:29:17.100400Z For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration.
DEFAULT 2023-06-23T10:29:17.100409Z at ChromeLauncher.resolveExecutablePath (/workspace/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ProductLauncher.js:304:27)
DEFAULT 2023-06-23T10:29:17.100417Z at ChromeLauncher.executablePath (/workspace/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ChromeLauncher.js:180:25)
DEFAULT 2023-06-23T10:29:17.100425Z at ChromeLauncher.computeLaunchArguments (/workspace/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ChromeLauncher.js:97:37)
DEFAULT 2023-06-23T10:29:17.100432Z at async ChromeLauncher.launch (/workspace/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ProductLauncher.js:83:28)
`

also tried down grading to version 18.1.0 but no luck so far

`DEFAULT 2023-06-23T12:07:36.668681Z error Error: Failed to launch the browser process! spawn /usr/bin/google-chrome ENOENT
DEFAULT 2023-06-23T12:07:36.668716Z TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md
DEFAULT 2023-06-23T12:07:36.668725Z at onClose (/workspace/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:299:20)
DEFAULT 2023-06-23T12:07:36.668731Z at ChildProcess.<anonymous> (/workspace/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:293:24)
DEFAULT 2023-06-23T12:07:36.668737Z at ChildProcess.emit (node:events:513:28)
DEFAULT 2023-06-23T12:07:36.668743Z at ChildProcess._handle.onexit (node:internal/child_process:289:12)
DEFAULT 2023-06-23T12:07:36.668750Z at onErrorNT (node:internal/child_process:476:16)
DEFAULT 2023-06-23T12:07:36.668757Z at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
`

**Comment by mauricio-gg** [comment]
I'm having the same issue. Why is this closed?

**Comment by sorinfratila** [comment]
still experiencing this as well: `Could not find Chrome (ver. 114.0.5735.90)`

**Comment by cc-thingo** [comment]
I'm having the same issue. `Could not find Chrome (ver. 114.0.5735.90)`

**Comment by mauricio-gg** [comment]
I was able to fix this by placing `.pupeteerrc.cjs` in my project's root directory

**Comment by OrKoN** [comment]
Please read the documentation and make sure when you deploy to the server that you install the binaries for the server operating system. If you believe there is a bug, please open a new issue with reproduction steps demonstrating that the browser is not downloaded to the cache directory after an `npm install` is run.

**Comment by sorinfratila** [comment]
> I was able to fix this by placing .pupeteerrc.cjs in my project's root directory

@mauricio-gg could you tell me what does that file contain exactly ?
also what version of puppeteer are you using ? are you importing from puppeteer or puppeteer-core ?

**Comment by sorinfratila** [comment]
locally this thing works for me, it's in Docker that it doesn't work.

**Comment by cc-thingo** [comment]
I already add `.pupeteerrc.cjs` in project. Locally this thing works for me, but it doesn't work in digital ocean server.

**Comment by sk8killer** [comment]
Same error for me with Firebase Hosting after deploying since yesterday : `Could not find Chrome (ver. 114.0.5735.133)`
I tried everything, but nothing works : 
- operating a clean install of all packages with :
`npm clean-install`
- cleaning the npm cahe with : 
`npm cache clean --force`
- adding a gcp-build script to install chrome after deployment on Firebase Hosting with this : 
`"gcp-build": "node node_modules/puppeteer/install.js"` (doens't work because node_modules doesn't seem to be accessible from the root hosting directory...)
- adding a `puppeteer.config.js` as detailed in the puppeteer docs : 
[https://pptr.dev/guides/configuration/#changing-the-default-cache-directory](https://pptr.dev/guides/configuration/#changing-the-default-cache-directory)

Unfortunately, i tried to downgrade puppeteer to the previews working version for me on Firebase Hosting wich was 10.4.0
and it still doesn't work.

This is very frustrating and urgent for me as part of the website i need to deploy is currently out of service !

I noticed that when I try to install puppeteer in local, the `.cache/puppeteer` folder does not install every time. On ten tries, the `.cache/puppeteer` directory has only been installed once.

So I don't think that it is a problem with Firebase Hosting but rather with Puppeteer or the Chrome install server.

This issue needs to be reopenned.

**Comment by besnikh** [comment]
I have been awake last 3 days with this :( 
Even ChatGPT couldn't help me :) 

These are my files that I managed to run Puppeteer on Docker and Google Cloud Run

Dockerfile

```
FROM node:latest

WORKDIR /app

COPY package.json yarn.lock ./

RUN yarn install

COPY . .

# Install puppeteer dependencies
RUN apt-get update && apt-get install -y \
    wget \
    gconf-service \
    libasound2 \
    libatk1.0-0 \
    libcairo2 \
    libcups2 \
    libfontconfig1 \
    libgdk-pixbuf2.0-0 \
    libgtk-3-0 \
    libnspr4 \
    libpango-1.0-0 \
    libxss1 \
    fonts-liberation \
    libappindicator1 \
    libnss3 \
    lsb-release \
    xdg-utils

ENV PUPPETEER_DOWNLOAD_PATH=/usr/local/chromium

RUN yarn add puppeteer

EXPOSE 4000

CMD ["yarn", "start"]
```

Some parts of index.js 

```
const browser = await puppeteer.launch({
    headless: "new",
    args: [
      '--no-sandbox',
      '--disable-dev-shm-usage'
    ],
  });

await page.goto(url, { waitUntil: 'domcontentloaded' });
```

Hope this helps someone but I don't know what happened with that Chromium it killed me

**Comment by sk8killer** [comment]
> Same error for me with Firebase Hosting after deploying since yesterday : `Could not find Chrome (ver. 114.0.5735.133)` I tried everything, but nothing works :
> 
> * operating a clean install of all packages with :
>   `npm clean-install`
> * cleaning the npm cahe with :
>   `npm cache clean --force`
> * adding a gcp-build script to install chrome after deployment on Firebase Hosting with this :
>   `"gcp-build": "node node_modules/puppeteer/install.js"` (doens't work because node_modules doesn't seem to be accessible from the root hosting directory...)
> * adding a `puppeteer.config.js` as detailed in the puppeteer docs :
>   https://pptr.dev/guides/configuration/#changing-the-default-cache-directory
> 
> Unfortunately, i tried to downgrade puppeteer to the previews working version for me on Firebase Hosting wich was 10.4.0 and it still doesn't work.
> 
> This is very frustrating and urgent for me as part of the website i need to deploy is currently out of service !
> 
> I noticed that when I try to install puppeteer in local, the `.cache/puppeteer` folder does not install every time. On ten tries, the `.cache/puppeteer` directory has only been installed once.
> 
> So I don't think that it is a problem with Firebase Hosting but rather with Puppeteer or the Chrome install server.
> 
> This issue needs to be reopenned.

UPDATE 29/06/2023 : Everything works now like a charm today... I don't know why !

**Comment by sorinfratila** [comment]
> I don't know why !

@sk8killer some people have commented that there might (have been/) be some issues with the server chrome binary is being downloaded from.

**Comment by tungvtrelipasoft** [comment]
Run node node_modules/puppeteer/install.js to download the binary

**Comment by tungvtrelipasoft** [comment]
Run node_modules/puppeteer/install.js to download the binary

**Comment by odebroqueville** [comment]
I have the same problem:

> error: "Could not find Chrome (ver. 114.0.5735.133). This can occur if either\n 1. you did not perform an installation before running the script (e.g. `npm install`) or\n 2. your cache path is incorrectly configured (which is: /root/.cache/puppeteer).\nFor (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration."

I also tried all of the measures below:

- operating a clean install of all packages with :
npm clean-install
- cleaning the npm cahe with :
npm cache clean --force
- adding a gcp-build script in package.json to install chrome after deployment on Firebase Hosting with this :
"gcp-build": "node node_modules/puppeteer/install.js" (doens't seem to do anything!)
- increasing the allocated memory to 1024M
- adding a puppeteer.config.js as detailed in the puppeteer docs :
https://pptr.dev/guides/configuration/#changing-the-default-cache-directory

but nothing worked! 

I'm not sure if the .puppeteer.config.cjs file should be placed in the root directory or in the functions folder:

![Screenshot 2023-07-03 at 19 08 29](https://github.com/puppeteer/puppeteer/assets/4818315/dbdb747f-fad5-481d-93ce-5e1a1a350412)

**Comment by joel-ghats** [comment]
> I have been awake last 3 days with this :( Even ChatGPT couldn't help me :)
> 
> These are my files that I managed to run Puppeteer on Docker and Google Cloud Run
> 
> Dockerfile
> 
> ```
> FROM node:latest
> 
> WORKDIR /app
> 
> COPY package.json yarn.lock ./
> 
> RUN yarn install
> 
> COPY . .
> 
> # Install puppeteer dependencies
> RUN apt-get update && apt-get install -y \
>     wget \
>     gconf-service \
>     libasound2 \
>     libatk1.0-0 \
>     libcairo2 \
>     libcups2 \
>     libfontconfig1 \
>     libgdk-pixbuf2.0-0 \
>     libgtk-3-0 \
>     libnspr4 \
>     libpango-1.0-0 \
>     libxss1 \
>     fonts-liberation \
>     libappindicator1 \
>     libnss3 \
>     lsb-release \
>     xdg-utils
> 
> ENV PUPPETEER_DOWNLOAD_PATH=/usr/local/chromium
> 
> RUN yarn add puppeteer
> 
> EXPOSE 4000
> 
> CMD ["yarn", "start"]
> ```
> 
> Some parts of index.js
> 
> ```
> const browser = await puppeteer.launch({
>     headless: "new",
>     args: [
>       '--no-sandbox',
>       '--disable-dev-shm-usage'
>     ],
>   });
> 
> await page.goto(url, { waitUntil: 'domcontentloaded' });
> ```
> 
> Hope this helps someone but I don't know what happened with that Chromium it killed me


 this worked for me 
 
 `FROM node:18

WORKDIR app

COPY package.json .
COPY package-lock.json .
RUN npm install

# Install necessary dependencies for Puppeteer
RUN apt-get update \
    && apt-get install -y wget gnupg \
    && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
    && sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' \
    && apt-get update \
    && apt-get install -y google-chrome-stable libxss1 \
    libx11-xcb1 \
    libxtst6 \
    libnss3 \
    libxss1 \
    libasound2 \
    libatk-bridge2.0-0 \
    libgtk-3-0 \
    --no-install-recommends \
    && rm -rf /var/lib/apt/lists/*

COPY . .

EXPOSE 8080

CMD ["npm", "start"]
`

**Comment by geminiyellow** [comment]
I'm having the same issue. `error Could not find Chrome (ver. 114.0.5735.133). `

**Comment by slochten** [comment]
Same issue here: `Could not find Chrome (ver. 114.0.5735.133)`

**Comment by goodalex** [comment]
Same issue: Could not find Chrome (ver. 114.0.5735.133)

**Comment by YuWenZhong** [comment]
The problem I encountered:
Using version 20, I downloaded a zip file in the $HOME/. cache/puppeteer/chrome folder, but the file was not successfully extracted. I attempted to manually extract it but also failed.

Guess:
There is an issue downloading the compressed package,  [Download link](https://goo.gle/chrome-for-testing) in [github document](https://github.com/puppeteer/puppeteer/tree/main)  I am unable to access and feel puzzled.

Solution:
I have specified to install version 19 instead of the default version 20.

`yarn add puppeteer@19`

**Comment by gabrielo91** [comment]
We are facing the same issue, we are using node 18, the project runs well locally, however we see that problem when we run puppeteer into a lambda:

```
Could not find Chrome (ver. 114.0.5735.133). This can occur if either 1. you did not perform an installation before running the script (e.g. `npm install`) or 2. your cache path is incorrectly configured (which is: /home/sbx_user1051/.cache/puppeteer).For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration.    at ChromeLauncher.resolveExecutablePath (/var/task/main.js:277082:27)    at ChromeLauncher.executablePath (/var/task/main.js:261665:25)    at ChromeLauncher.computeLaunchArguments (/var/task/main.js:261582:37)    at async ChromeLauncher.launch (/var/task/main.js:276861:28) | 2023-07-15T23:05:17.367Z cf293842-8d24-48e0-9826-26bdf4e787cc INFO error-handler-info non api error : Error: Could not find Chrome (ver. 114.0.5735.133). This can occur if either 1. you did not perform an installation before running the script (e.g. `npm install`) or 2. your cache path is incorrectly configured (which is: /home/sbx_user1051/.cache/puppeteer). For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration. at ChromeLauncher.resolveExecutablePath (/var/task/main.js:277082:27) at ChromeLauncher.executablePath (/var/task/main.js:261665:25) at ChromeLauncher.computeLaunchArguments (/var/task/main.js:261582:37) at async ChromeLauncher.launch (/var/task/main.js:276861:28)
-- | --
```

**Comment by TrustyTechSG** [comment]
Same issue here: Could not find Chrome (ver. 114.0.5735.133). works all fine on local development, but when deploy to Google cloud function (node 18), the error appears.

**Comment by vvivan89** [comment]
Facing the same issue with version114.0.5735.133
Had to change downloaded version using "browserRevision" to the .puppeteerrc.cjs:
`browserRevision: '117.0.5897.0'`

The latest downloadable version could be found [here](https://googlechromelabs.github.io/chrome-for-testing/known-good-versions-with-downloads.json).

**Comment by ciubakkaa** [comment]
> I have the same problem:
> 
> > error: "Could not find Chrome (ver. 114.0.5735.133). This can occur if either\n 1. you did not perform an installation before running the script (e.g. `npm install`) or\n 2. your cache path is incorrectly configured (which is: /root/.cache/puppeteer).\nFor (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration."
> 
> I also tried all of the measures below:
> 
> * operating a clean install of all packages with :
>   npm clean-install
> * cleaning the npm cahe with :
>   npm cache clean --force
> * adding a gcp-build script in package.json to install chrome after deployment on Firebase Hosting with this :
>   "gcp-build": "node node_modules/puppeteer/install.js" (doens't seem to do anything!)
> * increasing the allocated memory to 1024M
> * adding a puppeteer.config.js as detailed in the puppeteer docs :
>   https://pptr.dev/guides/configuration/#changing-the-default-cache-directory
> 
> but nothing worked!
> 
> I'm not sure if the .puppeteer.config.cjs file should be placed in the root directory or in the functions folder:
> 
> ![Screenshot 2023-07-03 at 19 08 29](https://user-images.githubusercontent.com/4818315/250625372-dbdb747f-fad5-481d-93ce-5e1a1a350412.jpg)
> 
> EDIT: I changed the filename to puppeteer.config.cjs and placed it in the functions folder. After deploying, a .cache folder was created with a very large number of files. `firebase deploy` ended with:
> 
> > ⚠  functions: Upload Error: HTTP Error: 400, `EntityTooLarge`Your proposed upload is larger than the maximum object size specified in your Policy Document.Content-length exceeds upper bound on range

I've ended up with the same size issue.
Bitbucket pipelines throws this error: `Error: HTTP Error: 400, <?xml version='1.0' encoding='UTF-8'?><Error><Code>EntityTooLarge</Code><Message>Your proposed upload is larger than the maximum object size specified in your Policy Document.</Message><Details>Content-length exceeds upper bound on range</Details></Error>`

**Comment by kreynen** [comment]
Running into the same issue. This is my first time installing Puppeteer. When I check `chromium-browser --product-version` it returns 115.0.5790.102, but trying to run the script calling Puppeteer returns an error that includes `Could not find Chrome (ver. 115.0.5790.98)`. I tried adding a .puppeteerrc.cjs like @vvivan89 suggested. I uninstalled and reinstalled Puppeteer with npm, but I'm still getting the same error.

**Comment by specinov-antoine** [comment]
I haved the same issue on Windows Server 2019 with IISNode and I finally solved it.
On my side it was linked to missing access rights on "C:\Windows\System32\config\systemprofile\.cache" directory for Windows user executing my IISNode application Pool.

**Comment by jmfadeley** [comment]
What a coincidence... ver. 115.0.5790.98.

**Comment by jmfadeley** [comment]
Discovery: The cache in `...\.cache\puppeteer\chrome` can be erroneous or not update. Simple fix is to go in, delete the old version, then do the `node node_modules\puppeteer\install.js` to pull it down. Worked aftewards.

**Comment by ReinhardRanner** [comment]
> Discovery: The cache in `...\.cache\puppeteer\chrome` can be erroneous or not update. `install.js` might claim the version you need is installed, but the datetime stamp didn't change. I got suspicious, so I deleted the old version, then ran `node node_modules\puppeteer\install.js` to pull it down again. My problem was fixed afterwards.

Yes that worked. Delete the current version from cache and run
`node node_modules\puppeteer\install.mjs` (Extension was mjs, not js in my case)

**Comment by dgtlmoon** [comment]
linux/ubuntu here - after `npm i puppeteer` I ran `node node_modules/puppeteer/install.mjs`

```
dgtlmoon:~$ node node_modules/puppeteer/install.mjs
Downloading chrome r117.0.5938.149 - 149.1 MB [====================] 100% 0.0s 
Chrome (117.0.5938.149) downloaded to /home/dgtlmoon/.cache/puppeteer/chrome/linux-117.0.5938.149
```

All good :)

**Comment by FlorianCostamagna** [comment]
> Run node_modules/puppeteer/install.js to download the binary

This works very well => in my case is was "node_modules/puppeteer/install.mjs" 
Thank you so much

**Comment by jakubcin** [comment]
> I have the same problem:
> 
> > error: "Could not find Chrome (ver. 114.0.5735.133). This can occur if either\n 1. you did not perform an installation before running the script (e.g. `npm install`) or\n 2. your cache path is incorrectly configured (which is: /root/.cache/puppeteer).\nFor (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration."
> 
> I also tried all of the measures below:
> 
> * operating a clean install of all packages with :
>   npm clean-install
> * cleaning the npm cahe with :
>   npm cache clean --force
> * adding a gcp-build script in package.json to install chrome after deployment on Firebase Hosting with this :
>   "gcp-build": "node node_modules/puppeteer/install.js" (doens't seem to do anything!)
> * increasing the allocated memory to 1024M
> * adding a puppeteer.config.js as detailed in the puppeteer docs :
>   https://pptr.dev/guides/configuration/#changing-the-default-cache-directory
> 
> but nothing worked!
> 
> I'm not sure if the .puppeteer.config.cjs file should be placed in the root directory or in the functions folder:
> 
> ![Screenshot 2023-07-03 at 19 08 29](https://user-images.githubusercontent.com/4818315/250625372-dbdb747f-fad5-481d-93ce-5e1a1a350412.jpg)
> 
> EDIT: I changed the filename to puppeteer.config.cjs and placed it in the functions folder. After deploying, a .cache folder was created with a very large number of files. `firebase deploy` ended with:
> 
> > ⚠  functions: Upload Error: HTTP Error: 400, `EntityTooLarge`Your proposed upload is larger than the maximum object size specified in your Policy Document.Content-length exceeds upper bound on range

> > linux/ubuntu, node v16.20.2 here - after `npm i puppeteer` I ran `node node_modules/puppeteer/install.mjs`
> > ```
> > dgtlmoon:~$ node node_modules/puppeteer/install.mjs
> > Downloading chrome r117.0.5938.149 - 149.1 MB [====================] 100% 0.0s 
> > Chrome (117.0.5938.149) downloaded to /home/dgtlmoon/.cache/puppeteer/chrome/linux-117.0.5938.149
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > All good :)
> 
> Confirming this fix also.



> > I have the same problem:
> > > error: "Could not find Chrome (ver. 114.0.5735.133). This can occur if either\n 1. you did not perform an installation before running the script (e.g. `npm install`) or\n 2. your cache path is incorrectly configured (which is: /root/.cache/puppeteer).\nFor (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration."
> > 
> > 
> > I also tried all of the measures below:
> > 
> > * operating a clean install of all packages with :
> >   npm clean-install
> > * cleaning the npm cahe with :
> >   npm cache clean --force
> > * adding a gcp-build script in package.json to install chrome after deployment on Firebase Hosting with this :
> >   "gcp-build": "node node_modules/puppeteer/install.js" (doens't seem to do anything!)
> > * increasing the allocated memory to 1024M
> > * adding a puppeteer.config.js as detailed in the puppeteer docs :
> >   https://pptr.dev/guides/configuration/#changing-the-default-cache-directory
> > 
> > but nothing worked!
> > I'm not sure if the .puppeteer.config.cjs file should be placed in the root directory or in the functions folder:
> > ![Screenshot 2023-07-03 at 19 08 29](https://user-images.githubusercontent.com/4818315/250625372-dbdb747f-fad5-481d-93ce-5e1a1a350412.jpg)
> > EDIT: I changed the filename to puppeteer.config.cjs and placed it in the functions folder. After deploying, a .cache folder was created with a very large number of files. `firebase deploy` ended with:
> > > ⚠  functions: Upload Error: HTTP Error: 400, `EntityTooLarge`Your proposed upload is larger than the maximum object size specified in your Policy Document.Content-length exceeds upper bound on range
> 
> I've ended up with the same size issue. Bitbucket pipelines throws this error: `Error: HTTP Error: 400, <?xml version='1.0' encoding='UTF-8'?><Error><Code>EntityTooLarge</Code><Message>Your proposed upload is larger than the maximum object size specified in your Policy Document.</Message><Details>Content-length exceeds upper bound on range</Details></Error>`

In my case everything works perfectly when i first deploy the firebase function, but after the next deploy it throws error "Could not find Chrome.... ", and maybe it would happen even with a new instance of the function (not sure about that) My guess is that gcp-build is not always running so i tried running node node_modules/puppeteer/install.js in my function using fork().
After installing it "manually" it works again.

```
const install_puppeteer = async (): Promise<any> => {
  const cacheDirectory = join(process.cwd(), '.cache', 'puppeteer');
  const fs = require('fs');
  if (fs.existsSync(cacheDirectory)) {
    return;
  } else {
    const child = fork('node_modules/puppeteer/install.js')
    return await once(child, 'close');
  }
}
```

I would like to call this function after every deploy and not letting user wait for it. Is there a way to do it ?
```
await install_puppeteer();
```

**Comment by ELHart05** [comment]
go for 
`cd node_modules_puppeteer`
then
`node install.mjs`

**Comment by jakepgoldman** [comment]
How does `node node_modules/puppeteer/install.mjs` work when deployed? Do I put it into my package.json? I ran but not seeing an effect on the deployed GCP cloud function.

**Comment by henryonsoftware** [comment]
I'm still got the same problem

Node: v18.17.1

Already install the puppeteer by `npm install puppeteer` and run `cd node_modules && node puppeteer/install.mjs`
It returns `Chrome (119.0.6045.105) downloaded to /home/master/.cache/puppeteer/chrome/linux-119.0.6045.105`

I check the `~/.cache/puppeteer/chrome`, there is a file name linux-119.0.6045.105

**Comment by fasilv843** [comment]
> How does `node node_modules/puppeteer/install.mjs` work when deployed? Do I put it into my package.json? I ran but not seeing an effect on the deployed GCP cloud function.

I also facing the same issue. puppeteer works fine on my localhost. But when i deployed it on renderer, I got this error 

_Could not find Chrome (ver. 119.0.6045.105). This can occur if either 1. you did not perform an installation before running the script (e.g. `npx puppeteer browsers install chrome`) or 2. your cache path is incorrectly configured (which is: /opt/render/.cache/puppeteer). For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration._

can we create a fuction to run `node node_modules/puppeteer/install.mjs` from server?

**Comment by vstets23** [comment]
Same here, can confirm that install.mjs script doesn't help anymore.
Running AWS Lambda functions based on Docker container with this setup:

```
FROM public.ecr.aws/lambda/nodejs:16

RUN curl -O https://lambda-insights-extension.s3-ap-northeast-1.amazonaws.com/amazon_linux/lambda-insights-extension.rpm && \
    rpm -U lambda-insights-extension.rpm && \
    rm -f lambda-insights-extension.rpm

RUN yum install -y amazon-linux-extras
RUN amazon-linux-extras install epel -y

ADD https://dl.google.com/linux/direct/google-chrome-stable_current_x86_64.rpm chrome.rpm
RUN yum install -y ./chrome.rpm

COPY ./  ./
RUN ls
RUN npm install
RUN node node_modules/puppeteer/install.mjs

CMD [ "index.handler" ]


```

**Comment by henryonsoftware** [comment]
I don't use Lambda so I don't know if it work but I made it works on VPS and this is my answer: https://stackoverflow.com/questions/77779601/error-could-not-find-chrome-ver-119-0-6045-105-this-can-occur-if-either

**Comment by SmtOmega** [comment]
> I already add `.pupeteerrc.cjs` in project. Locally this thing works for me, but it doesn't work in digital ocean server.

where you able to resolve this issue on digitalocean? if yes what were the step you took...thanks

**Comment by SmtOmega** [comment]
> I don't use Lambda so I don't know if it work but I made it works on VPS and this is my answer: https://stackoverflow.com/questions/77779601/error-could-not-find-chrome-ver-119-0-6045-105-this-can-occur-if-either

do you have any idea on how to resolve this on digitalocean?

**Comment by henryonsoftware** [comment]
I don't use DO, did you install the dependencies and move the .cache folder to your project folder?

**Comment by SmtOmega** [comment]
> I don't use DO, did you install the dependencies and move the .cache folder to your project folder?

Yes I did, I followed the step outline on configuring it with on the root of my folder, this work perfectly on my local system, but on digital ocean it still keeps throwing the error of :
 Could not find Chrome (ver. 119.0.6045.105). This can occur if either
 1. you did not perform an installation before running the script (e.g. npx puppeteer browsers install chrome) or
 2. your cache path is incorrectly configured (which is: /workspace/.cache/puppeteer

**Comment by widjesh** [comment]
Hey everyone,

I managed to resolve this issue by following the instructions outlined in [this documentation](https://pptr.dev/#installation).

Here's a summary of the steps you need to take:

1. Create a file named `.puppeteerrc.cjs` in your root folder.
2. Add the following content to the file:

    ```javascript
    const { join } = require('path');

    /**
     * @type {import("puppeteer").Configuration}
     */
    module.exports = {
      // Changes the cache location for Puppeteer.
      cacheDirectory: join(__dirname, '.cache', 'puppeteer'),
    };
    ```

3. Deploy your changes.

For local testing purposes, simply comment out the content in `.puppeteerrc.cjs`.

Following these steps, I successfully tested the flow on my Google Cloud Run instance. This approach might also work for Lambdas or Cloud Functions.


Let me know should you need any help

**Comment by permlap-phola** [comment]
@widjesh Hi I have followed your steps and it is still not working
I have deployed my nestjs app to google run
I have added .puppeteerrc.cjs according of this [doc](https://pptr.dev/#installation)

**package.json**
`"puppeteer": "^22.2.0",`

**Dockerfile**
```
FROM node:20 AS builder

# Create app directory
WORKDIR /app

# A wildcard is used to ensure both package.json AND package-lock.json are copied
COPY package*.json ./

# Install app dependencies
RUN npm install
RUN npx puppeteer browsers install chrome
COPY prisma ./prisma/
RUN npx prisma generate
# Explicitly run Puppeteer's install script to download the Chromium binary
COPY . .


RUN npm run build

FROM node:20

COPY --from=builder /app/node_modules ./node_modules
COPY --from=builder /app/package*.json ./
COPY --from=builder /app/dist ./dist

CMD [ "npm", "run", "start:prod" ]
```

**cloud build log**
![image](https://github.com/puppeteer/puppeteer/assets/137972470/4dce7aa5-c466-430f-b3c5-7baf887a0131)

**cloud** run log
![image](https://github.com/puppeteer/puppeteer/assets/137972470/a87e5067-8f6e-4f81-b596-216c2d5a17ba)

**Comment by zachsj** [comment]
I just found this thread and I want in on this convo, I've been trying to get this to work for days on Heroku and I'm using next.js. I tried adding the .cjs file, uninstalled and reinstalled puppeteer.  Added jontewks buildpack, tried following his readme.  Nothing works, Chrome will not launch.  Driving me nuts.

**Comment by zachsj** [comment]
It looks like some people had success above but is it still working or was it temporary?

**Comment by CraigIntuety** [comment]
As @zachsj has mentioned, ive also tried all of these and nothing is working. I literally just have a basic instance of puppeteer but via wsl/docker. puppeteer.launch keeps returning could not find chrome.

**Comment by CraigIntuety** [comment]
Im not entirely sure if this was the correct way around this but, i added to my alpine docker container

`RUN apk add chromium`

and rebuilt the container.

Then 
`  const browser = await puppeteer.launch({
        executablePath: '/usr/bin/chromium-browser',
        args: [ '--disable-gpu', '--disable-setuid-sandbox', '--no-sandbox', '--no-zygote' ] 
    })`

**Comment by elmeerr** [comment]
I switched to Puppeteer because I was facing EXACTLY the same issue with Playwright...

it is crazy how NO ONE has an proper answer to explain what is going on. Everything works perfectly but from the moment this is supposed to run from a Docker container, BOOM...nothing works as expected. In the beginning I thought "well, I am not a infra / devops guy so my Docker knowledge is very limited so that's probably why I can't get it working" but after months researching and trying EVERY single solution (first from Playwright and now Puppeteer) I am convinced that not even the maintainers can really explain why it does not work.

A lot of tries, a lot of different situations but not actual: THIS IS THE WAY 😅 

So frustrating

**Comment by SmtOmega** [comment]
> I switched to Puppeteer because I was facing EXACTLY the same issue with Playwright...
> 
> it is crazy how NO ONE has an proper answer to explain what is going on. Everything works perfectly but from the moment this is supposed to run from a Docker container, BOOM...nothing works as expected. In the beginning I thought "well, I am not a infra / devops guy so my Docker knowledge is very limited so that's probably why I can't get it working" but after months researching and trying EVERY single solution (first from Playwright and now Puppeteer) I am convinced that not even the maintainers can really explain why it does not work.
> 
> A lot of tries, a lot of different situations but not actual: THIS IS THE WAY 😅
> 
> So frustrating
I was on this same frustrating route with Digital Ocean, for several weeks, the company was on my neck to deliver have to migrate the server to aws elastic beanstalk and started another round of troubleshooting finally got it to work on aws beanstalk through someone's help. if you can migrate to aws you should instead of being stalk with no end result

**Comment by permlap-phola** [comment]
I have finally installed headless Chrome  on docker and deploy to google cloud. Howerver, installing is so tricky, you can read the offical doc of how to installing in difference enviroment here https://pptr.dev/troubleshooting. To be honest, the information in this doc seem not working for me neither.

What is working for me when installing  headless Chrome on docker running on google run is https://stackoverflow.com/a/76501278 I decide to downgrade puppeteer to 17.1.3 because the lastest version seem to have a lot of requirement to install.

What I would recommand for anyone who must use puppeteer in your project, please check out https://www.browserless.io
they will host headless Chrome for you. They said **Free for non-commercial uses.** in thier github

**Comment by zachsj** [comment]
I actually got it to work at least as far as the chrome issue. I followed widjesh steps above to create the .cjs file.  I actually had tried this several days ago and it didn't work.  It may have worked this time after I did npm uninstall npm install and npm build I think it was, because I was getting some ENOENT error.  I also removed any mkdir ./.cache.... scripts from the .json file.  After creating the .cjs file I did npm uninstall puppeteer then npm install puppeteer and it created a .cache file in the main directory.

My deployed page only works halfway though, it scrapes a site but then when it attempts to upload a file to Cloudinary I get "Error: Must supply api_key" in the Heroku logs.  Even after I set up the config var connection btwn Cloudinary and Heroku.  It works fine on development side, it's only after deploy I have this problem.  And surprisingly I can't really find any posts on any forums with people having this issue.  More frustration.

**Comment by zachsj** [comment]
Base64 conversion was the last step.  I'm all set!

**Comment by b4thefomo** [comment]
I never post but this is such a bitch that I have to. 

Can confirm that @widjesh method works, only thing I will add is for those on render add the following scripts to your package.json

`  "scripts": {
    "preinstall": "npm install puppeteer-core",
    "postinstall": "npx puppeteer browsers install chrome"
  },`

**Comment by gruckion** [comment]
> I'm seeing this too, was working fine but just stopped working on a fresh install. Does this always pull the latest chrome? Seen issues with this and Cypress before

Same here, I'm debugging trying to figure out why it works fine in a fresh turbo mono repo (nextJS/Turbo repo). But if I put that into my existing mono repo it worked first time. Then I added all my deps from the original app and it stopped working. So I change the deps back to when it worked and it still doesn't work.

I removed the node_modules and went back to the original version still not working 🤷 .

Still debugging, will message once I've figure it out.

**Comment by Vanortton** [comment]
> I never post but this is such a bitch that I have to.
> 
> Can confirm that @widjesh method works, only thing I will add is for those on render add the following scripts to your package.json
> 
> ` "scripts": { "preinstall": "npm install puppeteer-core", "postinstall": "npx puppeteer browsers install chrome" },`

Is it necessary to run these commands?

**Comment by SiwattDev** [comment]
Consegui resolver esse problema com essa resposta:
https://stackoverflow.com/a/76501278/20248141

Depois disso funcionou no Google Cloud Run.

**Comment by Vanortton** [comment]
I managed to solve this problem with this answer:
https://stackoverflow.com/a/76501278/20248141

After that it worked on Google Cloud Run.

**Comment by pontiggia** [comment]
> > I don't use DO, did you install the dependencies and move the .cache folder to your project folder?
> 
> Yes I did, I followed the step outline on configuring it on the root of my folder, this work perfectly on my local system, but digital ocean it still keeps throwing the error of : Could not find Chrome (ver. 119.0.6045.105). This can occur if either
> 
> 1. you did not perform an installation before running the script (e.g. npx puppeteer browsers install chrome) or
> 2. your cache path is incorrectly configured (which is: /workspace/.cache/puppeteer

Hey, Same problem and DO support sucks, I deployed from docker image, deployed from github, everything and the same error. HOW CAN I RESOLVE IT??

**Comment by SiwattDev** [comment]
> > > I don't use DO, did you install the dependencies and move the .cache folder to your project folder?
> > 
> > 
> > Yes I did, I followed the step outline on configuring it on the root of my folder, this work perfectly on my local system, but digital ocean it still keeps throwing the error of : Could not find Chrome (ver. 119.0.6045.105). This can occur if either
> > 
> > 1. you did not perform an installation before running the script (e.g. npx puppeteer browsers install chrome) or
> > 2. your cache path is incorrectly configured (which is: /workspace/.cache/puppeteer
> 
> Hey, Same problem and DO support sucks, I deployed from docker image, deployed from github, everything and the same error. HOW CAN I RESOLVE IT??

Use this docker image:
Dockerfile:
# Ubuntu 22.04 has support for many frameworks
FROM ubuntu:22.04

# Install Node & NPM
RUN apt-get update && apt-get upgrade -y
RUN apt-get install -y curl
RUN curl -sL https://deb.nodesource.com/setup_16.x | bash
RUN apt-get install -y nodejs


# Install Google Chrome
RUN apt install chromium-browser -y
RUN apt update 
# Install Puppeteer
RUN npm install -g puppeteer --unsafe-perm=true -allow-root
RUN apt install -y gconf-service libgbm-dev libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget


WORKDIR /usr/src/app
COPY . .
ENV NODE_ENV=production
RUN npm install --production


EXPOSE 80
CMD ["npm", "start"]

**Comment by Vanortton** [comment]
> > > I don't use DO, did you install the dependencies and move the .cache folder to your project folder?
> > 
> > 
> > Yes I did, I followed the step outline on configuring it on the root of my folder, this work perfectly on my local system, but digital ocean it still keeps throwing the error of : Could not find Chrome (ver. 119.0.6045.105). This can occur if either
> > 
> > 1. you did not perform an installation before running the script (e.g. npx puppeteer browsers install chrome) or
> > 2. your cache path is incorrectly configured (which is: /workspace/.cache/puppeteer
> 
> Hey, Same problem and DO support sucks, I deployed from docker image, deployed from github, everything and the same error. HOW CAN I RESOLVE IT??

Use this docker image:
Dockerfile:
# Ubuntu 22.04 has support for many frameworks
FROM ubuntu:22.04

# Install Node & NPM
RUN apt-get update && apt-get upgrade -y
RUN apt-get install -y curl
RUN curl -sL https://deb.nodesource.com/setup_16.x | bash
RUN apt-get install -y nodejs


# Install Google Chrome
RUN apt install chromium-browser -y
RUN apt update 
# Install Puppeteer
RUN npm install -g puppeteer --unsafe-perm=true -allow-root
RUN apt install -y gconf-service libgbm-dev libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget


WORKDIR /usr/src/app
COPY . .
ENV NODE_ENV=production
RUN npm install --production


EXPOSE 80
CMD ["npm", "start"]

**Comment by ryan-williams** [comment]
I have a Github Action that keeps hitting this error, in a strange pattern:
- initial, successful run
- re-run fails a few minutes later
- failure persists across retries
- job gets "unstuck" by changing the puppeteer version pin.

Best guess: [pnpm cache](https://github.com/neighbor-ryan/ctbk.dev/blob/aaa285981e408a7c6e939ce96b0a53be9aa2a42c/.github/workflows/www.yml#L23) is interacting with something that assumes the most recent Chrome version.

[Here's a succesful run](https://github.com/neighbor-ryan/ctbk.dev/actions/runs/9026757747/job/24804557222) from 15mins ago, which included [debug info](https://github.com/neighbor-ryan/ctbk.dev/actions/runs/9026757747/job/24804557222#step:9:1):

```bash
tree -L 3 ~/.cache/puppeteer
# /home/runner/.cache/puppeteer
# ├── chrome
# │   ├── linux-123.0.6312.105
# │   │   └── chrome-linux64
# │   └── linux-124.0.6367.91
# │       └── chrome-linux64
# └── chrome-headless-shell
#     ├── linux-123.0.6312.105
#     │   └── chrome-headless-shell-linux64
#     └── linux-124.0.6367.91
#         └── chrome-headless-shell-linux64
```

Same workflow [failed](https://github.com/neighbor-ryan/ctbk.dev/actions/runs/9026775212/job/24804652344) a few minutes later. That job only output:
```bash
tree -L 3 ~/.cache/puppeteer
# /home/runner/.cache/puppeteer
# └── chrome
#     └── linux-124.0.6367.91
#         └── chrome-linux64
```
before [failing with](https://github.com/neighbor-ryan/ctbk.dev/actions/runs/9026775212/job/24804652344#step:10:8):
```
/home/runner/work/ctbk.dev/ctbk.dev/www/node_modules/.pnpm/puppeteer-core@22.6.3/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ProductLauncher.js:288
                    throw new Error(`Could not find Chrome (ver. ${this.puppeteer.browserRevision}). This can occur if either\n` +
                          ^
Error: Could not find Chrome (ver. 123.0.6312.[10](https://github.com/neighbor-ryan/ctbk.dev/actions/runs/9026775212/job/24804652344#step:10:11)5). This can occur if either
 1. you did not perform an installation before running the script (e.g. `npx puppeteer browsers install chrome`) or
 2. your cache path is incorrectly configured (which is: /home/runner/.cache/puppeteer).
```

I [added](https://github.com/neighbor-ryan/ctbk.dev/commit/aaa285981e408a7c6e939ce96b0a53be9aa2a42c) a `pnpm up puppeteer`, and [a subsequent re-run](https://github.com/neighbor-ryan/ctbk.dev/actions/runs/9026879477/job/24804847946) worked, printing:
```bash
tree -L 3 ~/.cache/puppeteer
# /home/runner/.cache/puppeteer
# ├── chrome
# │   └── linux-124.0.6367.91
# │       └── chrome-linux64
# └── chrome-headless-shell
#     └── linux-124.0.6367.91
#         └── chrome-headless-shell-linux64
```

## Previous occurrence
This job only runs once a month, but it runs twice in quick succession when it runs. Last month I saw the same thing:
- [initial success](https://github.com/neighbor-ryan/ctbk.dev/actions/runs/8594866679)
- [failure](https://github.com/neighbor-ryan/ctbk.dev/actions/runs/8594984760/job/23548934610) a few minutes later
- [success](https://github.com/neighbor-ryan/ctbk.dev/actions/runs/8595060461) after [pinning puppeteer](https://github.com/neighbor-ryan/ctbk.dev/commit/f2c2ebbc5a66d57b840d8c601a153cf68a3bb28f) to a new version (in this case, 22.0.0 → 22.6.3).

**Comment by tonnyganteng** [comment]
just npm uninstall puppeteer and then reinstall it npm i puppeteer

**Comment by Tsgcode1122** [comment]
> run

thanks, this works for me, i host on render

**Comment by fsultani** [comment]
Figured I'd share the solution that worked for me, most of which is very similar to what others have already shared.

Create a [`puppeteer.config.cjs`](https://pptr.dev/guides/configuration#changing-the-default-cache-directory) file in the root directory of your project:

```
const { join } = require('path');

/**
 * @type {import("puppeteer").Configuration}
 */
module.exports = {
  // Changes the cache location for Puppeteer.
  cacheDirectory: join(__dirname, '.cache', 'puppeteer'),
};
```

For Render, this wasn't enough because after every new deployment, I'd get the same error that Chrome could't be found.  [Using this answer](https://stackoverflow.com/a/78809051), I added a command in my package.json file to install the appropriate version of chrome before starting the server:

```
"scripts": {
  "start": "npx puppeteer browsers install chrome && node greenhouse/index.js"
},
```

Everything seems to be working now.

**Comment by patrickalima98** [comment]
> [`puppeteer.config.cjs`](https://pptr.dev/guides/configuration#changing-the-default-cache-directory)

Thanks @fsultani  the [puppeteer.config.cjs](https://pptr.dev/guides/configuration#changing-the-default-cache-directory) works for me in GitLab CI/CD :)

**Comment by qzw881130** [comment]
this is error that i ran into.
```
Could not find Chrome (ver. 128.0.6613.119). This can occur if either
1.you did not perform an installation before running the script (e.g. npx puppeteer browsers install ${browserType}) or
2.your cache path is incorrectly configured (which is: C:\Users\jjp.cache\puppeteer).
```

**I solved.**
Cause I need to download in script, not Terminal. so, 
```
import { downloadBrowsers } from "puppeteer/internal/node/install.js";

async myfunc(){
    await downloadBrowsers();
   //other codes.
}
```

`package.json`
```
"puppeteer": "^23.3.1",
```

**Comment by ryan-williams** [comment]
I still run into this regularly, after many months. For example, [this run](https://github.com/neighbor-ryan/ctbk.dev/actions/runs/11420365690) just passed, then [this run](https://github.com/neighbor-ryan/ctbk.dev/actions/runs/11420384138) right afterward failed.

I think what's happening is:

- I'm using PNPM, with `actions/setup-node` caching based on my `pnpm-lock.yaml`
- If the lockfile hasn't changed, the full Puppeteer install flow isn't executed, so I have Puppeteer files under `node_modules`, but `~/.cache/puppeteer` is ≈empty.

I've just attempted to work around it by adding [an `@actions/cache` step for `~/.cache/puppeteer`](https://github.com/neighbor-ryan/ctbk.dev/blob/9506f8d230a65c5d8c8ff195b5a76cbbfd1d8cec/.github/workflows/www.yml#L24-L28):

```yml
      - name: Cache Puppeteer
        uses: actions/cache@v4
        with:
          key: puppeteer
          path: ~/.cache/puppeteer
```

I won't know if this fixed it for a few weeks though.

**Comment by HarisHera** [comment]
The thing that worked for me, besides adding the .puppeteerrc.cjs file as stated in the documentation was adding a postinstall script inside of the package.json. 

`"postinstall": "npx puppeteer browsers install"`

This will ensure that this command runs automatically after the npm install script.

**Comment by ZeeshanAhmadKhalil** [comment]
I followed @fsultani 's solutiob but  I am getting the following error:

```
Node.js v20.12.2
> zigglebee-apis@0.0.1 start:prod
> npx puppeteer browsers install chrome && NODE_ENV=production node dist/src/main
node:internal/modules/cjs/loader:1146
 throw err;
 ^
Error: Cannot find module '../puppeteer.js'
Require stack:
- /var/app/current/node_modules/.bin/puppeteer
   at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
   at Module._load (node:internal/modules/cjs/loader:984:27)
   at Module.require (node:internal/modules/cjs/loader:1231:19)
   at require (node:internal/modules/helpers:179:18)
   at Object.<anonymous> (/var/app/current/node_modules/.bin/puppeteer:14:40)
   at Module._compile (node:internal/modules/cjs/loader:1369:14)
   at Module._extensions..js (node:internal/modules/cjs/loader:1427:10)
   at Module.load (node:internal/modules/cjs/loader:1206:32)
   at Module._load (node:internal/modules/cjs/loader:1022:12)
   at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12) {
 code: 'MODULE_NOT_FOUND',
 requireStack: [ '/var/app/current/node_modules/.bin/puppeteer' ]
}
```

**Comment by ZeeshanAhmadKhalil** [comment]
This is how my issue was solved, had to install it manually on Amazon Linux 2023.
https://github.com/amazonlinux/amazon-linux-2023/discussions/417#discussioncomment-6724153

**Comment by Mozalino** [comment]
I found a fix for this in my use case with heroku, i used the command heroku run 'find /app -type f -name "chrome" 2>/dev/null' --app *insert app name here*

found the path for chrome. 

Changed it in my script and then great success :) Hope it helps someone!
